srcpkgs/qt5/patches/qtnetwork-5.15.0-libressl.patch

```diff
@@ -338,3 +338,49 @@ index f35e0ba2..30097317 100644
 -- 
 2.25.0
 
+--- qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp	2020-05-11 10:15:08.000000000 +0200
++++ qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp	2020-07-01 17:06:34.471202351 +0200
+@@ -264,7 +264,8 @@
+ DEFINEFUNC5(int, EVP_CipherInit, EVP_CIPHER_CTX *ctx, ctx, const EVP_CIPHER *type, type, const unsigned char *key, key, const unsigned char *iv, iv, int enc, enc, return 0, return)
+ DEFINEFUNC6(int, EVP_CipherInit_ex, EVP_CIPHER_CTX *ctx, ctx, const EVP_CIPHER *cipher, cipher, ENGINE *impl, impl, const unsigned char *key, key, const unsigned char *iv, iv, int enc, enc, return 0, return)
+ DEFINEFUNC5(int, EVP_CipherUpdate, EVP_CIPHER_CTX *ctx, ctx, unsigned char *out, out, int *outl, outl, const unsigned char *in, in, int inl, inl, return 0, return)
+-DEFINEFUNC3(int, EVP_CipherFinal, EVP_CIPHER_CTX *ctx, ctx, unsigned char *out, out, int *outl, outl, return 0, return)
++DEFINEFUNC3(int, EVP_CipherFinal_ex, EVP_CIPHER_CTX *ctx, ctx, unsigned char *out, out, int *outl, outl, return 0, return)
++DEFINEFUNC(int, EVP_CIPHER_CTX_cleanup, EVP_CIPHER_CTX *ctx, ctx, return 0, return)
+ DEFINEFUNC(const EVP_MD *, EVP_get_digestbyname, const char *name, name, return nullptr, return)
+ #ifndef OPENSSL_NO_DES
+ DEFINEFUNC(const EVP_CIPHER *, EVP_des_cbc, DUMMYARG, DUMMYARG, return nullptr, return)
+@@ -962,7 +962,8 @@
+     RESOLVEFUNC(EVP_CipherInit)
+     RESOLVEFUNC(EVP_CipherInit_ex)
+     RESOLVEFUNC(EVP_CipherUpdate)
+-    RESOLVEFUNC(EVP_CipherFinal)
++    RESOLVEFUNC(EVP_CipherFinal_ex)
++    RESOLVEFUNC(EVP_CIPHER_CTX_cleanup)
+     RESOLVEFUNC(EVP_get_digestbyname)
+ #ifndef OPENSSL_NO_DES
+     RESOLVEFUNC(EVP_des_cbc)
+--- qtbase/src/network/ssl/qsslkey_openssl.cpp	2020-05-11 10:15:08.000000000 +0200
++++ qtbase/src/network/ssl/qsslkey_openssl.cpp	2020-07-01 17:58:44.674363835 +0200
+@@ -360,7 +360,8 @@
+     q_EVP_CipherUpdate(ctx,
+         reinterpret_cast<unsigned char *>(output.data()), &len,
+         reinterpret_cast<const unsigned char *>(data.constData()), data.size());
+-    q_EVP_CipherFinal(ctx,
++    q_EVP_CipherFinal_ex(ctx,
+         reinterpret_cast<unsigned char *>(output.data()) + len, &i);
++    q_EVP_CIPHER_CTX_cleanup(ctx);
```

Comment:
```
     EVP_CIPHER_CTX_reset() clears all information from ctx and frees all
     allocated memory associated with it, except the ctx object itself, such
     that it can be reused for another series of calls to EVP_CipherInit(),
     EVP_CipherUpdate(), and EVP_CipherFinal().  It is also suitable for
     cipher contexts on the stack that were used and are no longer needed.
     EVP_CIPHER_CTX_cleanup() is a deprecated alias for
     EVP_CIPHER_CTX_reset().
```
There is a `_reset()` right after this, so `_cleanup()` shouldn't be needed.
(Also in the libressl code, `_reset()` is actually the alias, but nevertheless we don't need to call it twice)

Reply:
Also I'm quite sure we could just ignore the warning. It is mainly about the fact that `EVP_CipherFinal` used to cleanup itself and the warning is just about people that might use it without cleaning up. The only difference between `EVP_CipherFinal_ex` and `EVP_CipherFinal` is that one has a `int ret` and the other one directly returns the value from called functions :D.
So imo just change q_EVP_CipherFinal -> q_EVP_CipherFinal_ex and everything else is not needed.